examples/pytorch/graphsaint/train_full.py

```diff
@@ -0,0 +1,168 @@
+import argparse
+import os
+import time
+import numpy as np
+import torch
+import torch.nn.functional as F
+from modules import GCNNet
+from utils import Logger, evaluate, save_log_dir, load_data
+
+
+def main(args):
+
+    multitask_data = set(['ppi', 'yelp', 'amazon'])
+    multitask = args.dataset in multitask_data
+
+    # load and preprocess dataset
+    data = load_data(args, multitask)
+    g = data.g
+    train_mask = g.ndata['train_mask']
+    val_mask = g.ndata['val_mask']
+    test_mask = g.ndata['test_mask']
+    labels = g.ndata['label']
+
+    train_nid = np.nonzero(train_mask.data.numpy())[0].astype(np.int64)
+    train_g = g.subgraph(train_nid)
```

Comment:
Have you checked if the relative order of the nodes is preserved? You can retrieve the original node IDs with `g.subgraph(train_nid, store_ids=True)`.